compiler/modules/types/list.bal

```diff
@@ -188,29 +182,52 @@ function listFormulaIsEmpty(Context cx, Conjunction? pos, Conjunction? neg) retu
     return !listInhabited(cx, members, rest, neg);
 }
 
+function listIntersectWith(FixedLengthArray members, SemType rest, ListAtomicType lt) returns [FixedLengthArray, SemType]? {
+    int ltLen = lt.members.fixedLength;
+    int newLen = int:max(members.fixedLength, ltLen);
+    if members.fixedLength < newLen {
```

Comment:
@jclark As an optimization we can handle `members.initial.length() == lt.members.initial.length()` case without pre-filling `members`.  Should we?

Reply:
Do whatever is least code first, and put in a comment about possible future optimization.